drivers/gpio/Kconfig.nrf5

```diff
@@ -14,6 +14,12 @@ menuconfig GPIO_NRF5
 
 if GPIO_NRF5
 
+config GPIO_NRF5_INIT_PRIORITY
```

Comment:
Why is this needed, please add a comment in the commit message explaining why we need to configure the init priority

Reply:
This isn't strictly needed, but I think it makes the board power rail init clearer. The issue is that the GPIO controller has to be initialized before the VDD rail can be powered up. Due to this dependency I felt that making the GPIO controller have its own init priority would make sense. I considered not doing this and explaining in the board Kconfig file where the rail initialization priorities are set that the BOARD_VDD_PWR_CTRL_INIT_PRIORITY has to be larger than KERNEL_INIT_PRIORITY_DEFAULT because that's when the GPIO controller is currently initialized. I could drop this commit and do that instead, if you think it's the better option.